airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,312 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains SFTP Batch operator."""
+import ast
+import os
+import re
+from pathlib import Path
+from typing import List, Optional
+
+from paramiko.sftp_client import SFTPClient
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.sftp.operators.sftp import SFTPOperation
+from airflow.providers.sftp.utils.common import check_conn, make_intermediate_dirs
+from airflow.utils.context import Context
```

Comment:
This import needs to be under `TYPE_CHECKING` otherwise this provider will have a minimum Airflow version dependency of 2.2.3. Check out an [example](https://github.com/apache/airflow/blob/82e466d8f9b196e1efba44cc15214b9d0bd3b2d1/airflow/providers/databricks/operators/databricks.py#L28-L29) in the Databricks operator file.

Then the `execute()` typing will be:
```python
def execute(self, context: "Context") -> None:
```